### PR TITLE
Role validations

### PR DESCRIPTION
**Pull request by GaryGapinski**
This request addresses issue #51 as well as other role-related constraints (the latter lack a specific issue).

### CODE REVIEW

**Comment by GaryGapinski** [comment]
This PR includes all role-related items other than <component> and <inventory-item> <responsible-party> and privacy PoC.

**Review by ohsh6o — COMMENTED** [review]
B: I am not sure how this works. I see a lot of important supporting work in this PR, but not where it trickles down into `responsible-role`s. Is that correct? From SSP Guide, Section 5.2, page 36.

```xml
   <metadata>
      <role id="admin-unix">
         <title>Unix Administrator</title>
      </role>
   </metadata>

  <!-- Fragment: -->
   <system-implementation>
      <user uuid="uuid-value">
         <role-id>admin-unix</role-id>
      </user>
   </system-implementation >

   <!-- system-implementation -->
   <control-implementation>
      <implemented-requirement uuid="uuid-value" control-id="ac-2">
         <!-- cut -->
         <responsible-role role-id="admin-unix" />
         <set-parameter param-id="ac-1_prm_a">
            <value>System Manager, System Architect, ISSO</value>
         </set-parameter >
         <!-- cut -->
   </control-implementation>
```

File: resources/validations/src/ssp.sch [hunk 1]
```diff
@@ -60,374 +78,505 @@
...
+        <xsl:param
+            name="level" />
+        <xsl:variable
+            name="profile-map">
             <!-- 
             OSCAL releases are tagged, but updates from OSCAL CI/CD pipeline to
             github.com/usnistgov/oscal-content are not. The 0f78f05 commit is the
             most recent triggered by the OSCAL 1.0.0-rc1 release. Change this url
             accordingly if you know what you are doing.
             -->
```
[the remark this answers is not in the document]
Comment (GaryGapinski) [reply resources/validations/src/ssp.sch]: SSP Guide, Section 5.2, page 36 is handled by [this](https://github.com/18F/fedramp-automation/blob/role-validations/resources/validations/src/ssp.sch#L1345-L1373) and [this](https://github.com/18F/fedramp-automation/blob/role-validations/resources/validations/src/ssp.sch#L1376-L1379).

If the **FedRAMP** resolved catalogs ("profiles") get changed they will be relatively referenced. I'll delete the comment.

[the remark this answers is not in the document]
Comment (ohsh6o) [reply resources/validations/src/ssp.sch]: Sorry, I am not sure what goes on with the diffs but they are always highlighting signficant formatting changes and I am missing obvious stuff.

I will re-review appropriately. Suppressing this comment.

File: resources/validations/test/ssp.xspec [hunk 2]
```diff
@@ -5188,4 +5188,427 @@
             </x:scenario>
         </x:scenario>
     </x:scenario>
+    <x:scenario
+        label="FedRAMP OSCAL SSP User-related unit tests">
+        <x:scenario
+            label="affirmative">
+            <x:context>
+                <ssp
```
[the remark this answers is not in the document]
Comment (GaryGapinski) [reply resources/validations/test/ssp.xspec:5196]: XSpec defines an artificial context. The context need only be adequate to the unit test task and the Schematron assertion context.

Note that the [assertions](https://github.com/18F/fedramp-automation/blob/role-validations/resources/validations/src/ssp.sch#L1374-L1451) have relative contexts: i.e., not rooted at the document node. The XSpec [context](https://github.com/18F/fedramp-automation/blob/role-validations/resources/validations/test/ssp.xspec#L5193-L5225) is the minimum necessary for those affirmative tests. The Schematron assertions do not care about the encompassing (larger) context at the time the XSpec delivers it. None of those assertions expect a specific root element.

[the remark this answers is not in the document]
Comment (ohsh6o) [reply resources/validations/test/ssp.xspec:5196]: You clean up the code a lot, so I guess this should be something I noticed loooooong ago!

**Review by ohsh6o — APPROVED** [review]
LGTM

**Pull request merged**